tests/run/carray_slicing.pyx

```diff
@@ -275,3 +277,214 @@ def struct_ptr_iter():
+        a[i] = i**2
+    b = a[8:]
+    return b
+
+cpdef list test_c_to_py_coercing_array_in_var_2():
+    """
+    >>> test_c_to_py_coercing_array_in_var_2()
+    [64, 81]
+    """
+    cdef int i
+    cdef int a[10]
+    b = 5555
+    for i in range(10):
+        a[i] = i**2
+    b = a[8:]
+    return b
+
+
+cpdef tuple test_coercing_c_slice_to_tuple():
+    """
+    >>> test_coercing_c_slice_to_tuple()
+    (64, 81)
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+    return a[8:]
+
+
+cpdef set test_coercing_c_slice_to_set():
+    """
+    >>> test_coercing_c_slice_to_set()
+    set([64, 81])
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+    return a[8:]
+
+
+cpdef frozenset test_coercing_c_slice_to_frozenset():
+    """
+    >>> test_coercing_c_slice_to_frozenset()
+    frozenset([64, 81])
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+    return a[8:]
+
+def test_coercing_c_slice_by_casting():
+    """
+    >>> test_coercing_c_slice_by_casting()
+    [25, 36, 49, 64, 81]
+    (25, 36, 49, 64, 81)
+    set([64, 25, 36, 81, 49])
+    frozenset([64, 25, 36, 81, 49])
+    """
+    cdef int i
+    cdef int a[10]
+    for i in range(10):
+        a[i] = i**2
+
+    print(<list> a[5:])
+    print(<tuple> a[5:])
+    print(<set> a[5:])
+    print(<frozenset> a[5:])
+
+
+
+
+
+# Error !!!!!!!!
+#def test_coercing_c_slice_by_casting():
+#    """
+#    >>> test_coercing_c_slice_by_casting()
+#    [25, 36, 49, 64, 81]
+#    (25, 36, 49, 64, 81)
+#    set([64, 25, 36, 81, 49])
+#    frozenset([64, 25, 36, 81, 49])
+#    """
+#    cdef int i
+#    cdef int a[10]
+#    b = 5555
+#    for i in range(10):
+#        a[i] = i**2
+#    b = a
+#    print(<list> b)
+#    print(<tuple> b)
+#    print(<set> b)
+#    print(<frozenset> b)
+#
+#    # return lists
+#    #[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]
+#    #[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]
+#    #[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]
+#    #[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]
+
+
+# Error !!!!!!!!
+#def test_coercing_c_slice_by_casting():
+#    """
+#    >>> test_coercing_c_slice_by_casting()
+#    [25, 36, 49, 64, 81]
+#    (25, 36, 49, 64, 81)
+#    set([64, 25, 36, 81, 49])
+#    frozenset([64, 25, 36, 81, 49])
+#    """
+#    cdef int i
+#    cdef int a[10]
+#    b = 5555
+#    for i in range(10):
+#        a[i] = i**2
+#    b = a[1:5]
+#    print(<list> b)
+#    print(<tuple> b)
+#    print(<set> b)
+#    print(<frozenset> b)
+#
+#    # return lists
+#    #[1, 4, 9, 16]
+#    #[1, 4, 9, 16]
+#    #[1, 4, 9, 16]
+#    #[1, 4, 9, 16]
+
+
+#error
+#def test_coercing_c_slice_by_casting():
+#    """
+#    >>> test_coercing_c_slice_by_casting()
+#    [25, 36, 49, 64, 81]
+#    (25, 36, 49, 64, 81)
+#    set([64, 25, 36, 81, 49])
+#    frozenset([64, 25, 36, 81, 49])
+#    """
+#    cdef int i
+#    cdef int a[10]
+     # is not initializaed
+#    #b = 5555
```

Comment:
If "b" isn't initialized ...

Reply:
See comment above - I don't think this is a bug. It just doesn't know that `b` should be an object.